### PR TITLE
add general purpose meter update method

### PR DESCRIPTION
**Pull request by copperlight**
Reduce cognitive load associated with picking the correct meter method to send a value to spectatord, while maintaining the type and value checks. This also helps simplify meta-programming tasks associated with building context managers and decorators.

### CODE REVIEW

File: spectator/meter/__init__.py [hunk 1]
```diff
@@ -12,3 +14,22 @@ def __init__(self, meter_id: MeterId, writer: WriterUnion, meter_type_symbol: st
...
+            line = f"{self._meter_type_symbol}:{self._id.spectatord_id}:{value}"
+            self._writer.write(line)
+        elif self._meter_type_symbol == "U" and isinstance(value, c_uint64):
+            # monotonic_counter_uint
+            line = f"{self._meter_type_symbol}:{self._id.spectatord_id}:{value.value}"
+            self._writer.write(line)
+        else:
+            # age_gauge, gauge, max_gauge, monotonic_counter
+            line = f"{self._meter_type_symbol}:{self._id.spectatord_id}:{value}"
+            self._writer.write(line)
```
Comment (amjith) [inline spectator/meter/__init__.py:35]: I'm a little unclear on how the `else` is different from the other sections. 

If someone calls `update(-5)` and the meter is a `dist_summary` the `elif` is fail but the `else` will end up publishing the negative value for a `dist_summary`.

Comment (copperlight) [reply spectator/meter/__init__.py:35]: Good point! Will fix.

**Comment by copperlight** [comment]
The key thing to think about - is this going to be useful and used, or it is better to drop it and keep the overall implementation as-is.

**Comment by copperlight** [comment]
We do not have a particularly good use case for this generic method name.

**Pull request closed without merging**